root/static/scripts/common/components/ReleaseEvents.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React, {useCallback, useState} from 'react';
+
+import CountryAbbr from '../../../../components/CountryAbbr';
+import hydrate from '../../../../utility/hydrate';
+import {bracketedText} from '../utility/bracketed';
+import formatDate from '../utility/formatDate';
+import isDateEmpty from '../utility/isDateEmpty';
+
+import EntityLink from './EntityLink';
+
+const COLLAPSE_THRESHOLD = 3;
+
+const releaseEventKey = event => (
+  String(event.country ? event.country.id : '') + '\0' +
+  formatDate(event.date)
+);
+
+const buildReleaseEventRow = (event, abbreviated) => {
+  const country = event.country;
+  const hasDate = !isDateEmpty(event.date);
+
+  return (
+    <li
+      aria-label={l('Release event')}
+      className="release-event"
+      key={releaseEventKey(event)}
+    >
+      {country ? (
+        abbreviated ? (
+          <CountryAbbr
+            className="release-country"
+            country={country}
+            withLink
+          />
+        ) : (
+          <EntityLink entity={country} />
+        )
+      ) : (
+        abbreviated ? (
+          <span
+            className="release-country no-country"
+            title={l('Missing country')}
+          >
+            {lp('-', 'missing data')}
+          </span>
+        ) : null
+      )}
+
+      {(abbreviated || !country || !hasDate) ? null : <br />}
+
+      {hasDate ? (
+        <span className="release-date">
+          {formatDate(event.date)}
+        </span>
+      ) : (
+        abbreviated ? (
+          <span
+            className="release-date no-date"
+            title={l('Missing date')}
+          >
+            {lp('-', 'missing data')}
+          </span>
+        ) : null
+      )}
+    </li>
+  );
+};
+
+type ReleaseEventsProps = {|
+  +abbreviated?: boolean,
+  +events: ?$ReadOnlyArray<ReleaseEventT>,
+|};
+
+const ReleaseEvents = ({
+  abbreviated = true,
+  events,
+}: ReleaseEventsProps) => {
+  const [expanded, setExpanded] = useState<boolean>(false);
+
+  const expand = useCallback(event => {
+    event.preventDefault();
+    setExpanded(true);
+  });
+
+  const collapse = useCallback(event => {
+    event.preventDefault();
+    setExpanded(false);
+  });
+
+  const containerProps = {
+    'aria-label': l('Release events'),
+    'className': 'release-events' +
+      (abbreviated ? ' abbreviated' : ' links'),
+  };
+
+  const tooManyEvents = events
+    ? events.length > COLLAPSE_THRESHOLD
+    : false;
+
+  return (
+    (events && events.length) ? (
+      <>
+        {(tooManyEvents && !expanded) ? (
+          <>
+            <ul {...containerProps}>
+              {events.slice(0, COLLAPSE_THRESHOLD - 1).map(
```

Comment:
The `li.show-all` element has larger height than `li.release-event`. Thus, if there are `COLLAPSE_THRESHOLD + 1` events, then the release event list takes more space collapsed than before this commit. One way out would be to ensure hiding at least 2 release events as follows:

```suggestion
              {events.slice(0, COLLAPSE_THRESHOLD - 2).map(
```

Reply:
I made those changes but also reduced the padding around those links a bit (from 1em to 0.75em), since it did seem a bit much.